code/modules/clothing/neck/necklace.dm

```diff
@@ -0,0 +1,78 @@
+/obj/item/clothing/neck/necklace
+	name = "necklace"
+	desc = "A simple necklace."
+	icon_state = "necklace"
+	item_state = "necklace"
+	item_color = "necklace"
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/clothing/neck/necklace/long
+	name = "large necklace"
+	desc = "A large necklace."
+	icon_state = "necklacelong"
+	item_state = "necklacelong"
+	item_color = "necklacelong"
+
+/obj/item/clothing/neck/necklace/dope
+	name = "gold necklace"
+	desc = "Damn, it feels good to be a gangster."
+	icon_state = "bling"
+	item_state = "bling"
+	item_color = "bling"
+
+/obj/item/clothing/neck/necklace/locket
+	name = "gold locket"
+	desc = "A gold locket that seems to have space for a photo within."
+	icon_state = "locketgold"
+	item_state = "locketgold"
+	item_color = "locketgold"
+	var/base_icon
+	var/open
+	var/obj/item/held //Item inside locket.
+
+/obj/item/clothing/neck/necklace/locket/Destroy()
+	QDEL_NULL(held)
+	return ..()
+
+
+/obj/item/clothing/neck/necklace/locket/attack_self__legacy__attackchain(mob/user as mob)
+	if(!base_icon)
+		base_icon = icon_state
+
+	if(!("[base_icon]_open" in icon_states(icon)))
+		to_chat(user, "[src] doesn't seem to open.")
+		return
+
+	open = !open
+	to_chat(user, "You flip [src] [open?"open":"closed"].")
+	if(open)
+		icon_state = "[base_icon]_open"
+		if(held)
+			to_chat(user, "[held] falls out!")
```

Comment:
I know these are probably more things that were already in place, but While You're here
```suggestion
			to_chat(user, "<span class='warning'>[held] falls out!</span>")
```